src/service/standard-actions-impl.js

```diff
@@ -441,6 +445,71 @@ export class StandardActions {
 
     return null;
   }
+
+  /**
+   * Handles "setCookie" action.
+   * @param {!./action-impl.ActionInvocation} invocation
+   * @return {?Promise}
+   * @private Visible to tests only.
+   */
+  handleSetCookie_(invocation) {
+    const {node, args} = invocation;
+    const win = (node.ownerDocument || node).defaultView;
+
+    // Use RFC 6265 for allowed cookie name and value characters
+
+    const name = user().assertString(
+      args['name'],
+      "Argument 'name' must be a string."
+    );
+    userAssert(
+      /^[a-z0-9!#$%&'*+\-.^_`|~]+$/i.test(name),
+      "Argument 'name' is required and must contain only characters allowed by RFC 6265"
+    );
+
+    const value = user().assertString(
+      args['value'],
+      "Argument 'value' must be a string."
+    );
+    userAssert(
+      /^[a-z0-9!#$%&'()*+\-./:<=>?@[\]^_`{|}~]*$/i.test(value),
+      "Argument 'value' is required and must contain only characters allowed by RFC 6265"
+    );
```

Comment:
Might be worthwhile to add a max length check: `(name + value).length < 500` for example. The max length should take into consideration typical browser limitations on _total_ cookie length per domain: https://stackoverflow.com/questions/640938/what-is-the-maximum-size-of-a-web-browsers-cookies-key